nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/dao/impl/StandardFlowRegistryDAO.java

```diff
@@ -116,12 +116,19 @@ public Set<FlowRegistryBranch> getBranchesForUser(final FlowRegistryClientUserCo
             if (flowRegistry == null) {
                 throw new IllegalArgumentException("The specified registry id is unknown to this NiFi.");
             }
+            final FlowRegistryBranch defaultBranch = flowRegistry.getDefaultBranch(context);
             final Set<FlowRegistryBranch> branches = flowRegistry.getBranches(context);
-            final Set<FlowRegistryBranch> sortedBranches = new TreeSet<>(Comparator.comparing(FlowRegistryBranch::getName));
+            // Sort the default branch first, this allows main to always be the first
+            final Set<FlowRegistryBranch> sortedBranches = new TreeSet<>((branch1, branch2) -> {
+                if (branch1.getName().equals(defaultBranch.getName())) {
+                    return -1;
+                }
```

Comment:
Is it enough to handle `branch1` being the default branch here?
Or should we handle the case that the default branch is provided as `branch2` as well?

```suggestion
                if (branch1.getName().equals(defaultBranch.getName())) {
                    return -1;
                } else if (branch2.getName().equals(defaultBranch.getName())) {
                    return 1;
                } 
```